x/plans/keeper/migrations.go

```diff
@@ -59,3 +60,21 @@ func (m Migrator) Migrate2to3(ctx sdk.Context) error {
 
 	return nil
 }
+
+// Migrate3to4 implements store migration from v3 to v4:
+// - Trigger the version upgrade of the planFS fixation store
+// - Replace the store prefix from module-name ("plan") to "plan-fs"
+func (m Migrator) Migrate3to4(ctx sdk.Context) error {
+	const V4_PlanFixationStorePrefix = "plan"
+
+	// Note: MigratePrefix() already calls MigrateVersion() and thus will perform
+	// FixationStore version migration if needed, using the old prefix first.
+	// (This is required because otherwise the usual fixation-store MigrateVersion
+	// would wrongly use the new prefix).
+
+	if err := m.keeper.plansFS.MigratePrefix(ctx, V4_PlanFixationStorePrefix); err != nil {
```

Comment:
why do we explicit call MigratePrefix and not the MigrateVertion?

Reply:
cant we migrate from the old prefix directly to the new prefix(with the key change)

Reply:
I will rename "MigratePrefix()" to "MigrateVersionWithPrefix()" to make it much clearer.